custom_reg_form/management/commands/email_opt_in_list_to_s3.py

```diff
@@ -0,0 +1,257 @@
+
+
+LOGGER = logging.getLogger(__name__)
+
+def get_registration_ext_model():
+    """Returns the REGISTRATION_EXTENSION_MODEL"""
+    # first attempt to infer from the Meta field in the form
+    if hasattr(settings, "REGISTRATION_EXTENSION_FORM"):
+        try:
+            form_class_path = settings.REGISTRATION_EXTENSION_FORM
+            module_path = ".".join(form_class_path.split(".")[:-1])
+            module = importlib.import_module(module_path)
+            form_class = getattr(module, form_class_path.split(".")[-1])
+            if issubclass(form_class, ModelForm):
+                model_class = form_class.Meta.model
+                return model_class
+        except Exception:   # pylint: disable=broad-except
+            # we ignore execptions so we can try to get the
+            # model class in another way
+            pass
+
+    # check REGISTRATOIN_EXTENSION_MODEL setting
+    if hasattr(settings, "REGISTRATION_EXTENSION_MODEL"):
+        try:
+            model_class_path = settings.REGISTRATION_EXTENSION_MODEL
+            module_path = ".".join(model_class_path.split(".")[:-1])
+            module = importlib.import_module(module_path)
+            model_class = getattr(module, model_class_path.split(".")[-1])
+            return model_class
+        except Exception:    # pylint: disable=broad-except
+            # ignore and try again
+            pass
+
+    # check REGISTRATION_EXTENSION_MODEL environment variable
+    if "REGISTRATION_EXTENSION_MODEL" in os.environ:
+        try:
+            model_class_path = os.environ.get("REGISTRATION_EXTENSION_MODEL")
+            module_path = ".".join(model_class_path.split(".")[:-1])
+            module = importlib.import_module(module_path)
+            model_class = getattr(module, model_class_path.split(".")[-1])
+            return model_class
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+    raise Exception("Couldn't infer the model path to the registration extension model.")
+
+def infer_field_model(field):
+    """Returns the model of which `field` is a member"""
+
+    # first check in the User model
+    if hasattr(User, field):
+        return User
+    # check if in the UserProfile model
+    if hasattr(UserProfile, field):
+        return UserProfile
+    # check if in the REGISTRAION_EXTENSION_MODEL
+    reg_ext_model_class = get_registration_ext_model()
+    if hasattr(reg_ext_model_class, field):
+        return reg_ext_model_class
+
+    raise Exception("Couldn't infer field: `{}` model.".format(field))
+
+def construct_query(fields):
+    """Returns a string `query` to be passed to `cursor.execute`"""
+    reg_ext_table = get_registration_ext_model()._meta.db_table # pylint: disable=protected-access
+    field_names = ", ".join(
+        [".".join([infer_field_model(field)._meta.db_table, field]) for field in fields]    # pylint: disable=protected-access
+    )
+    query = """
```

Comment:
Is there any reason using raw SQL queries over Django's ORM?

Reply:
Just following the same behavior through the rest of the codebase , like [here](https://github.com/edx/edx-ora2/blob/master/openassessment/data.py#L119), in this case i didn't specify any fixed length and followed the pep recommendation of letting the underlying database implementation decide.

Reply:
@gabor-boros , can I resolve this or should we re-write to use ORM? The command is working as intended, I ran it and found the file in s3.